### PR TITLE
add onSubmit on suggest search input to redirect

### PR DESCRIPTION
**Pull request by maudetes**
Fix https://github.com/etalab/data.gouv.fr/issues/371.

Any suggestion for a better pattern is welcome ! :)

### CODE REVIEW

File: theme/js/components/search/search-input.vue [hunk 1]
```diff
@@ -57,6 +58,7 @@ export default {
     stop: Function,
     value: String,
     placeholder: String,
+    redirectUrl: String,
```
Comment (abulte) [inline theme/js/components/search/search-input.vue:61]:
```suggestion
    submitUrl: String,
```

**Review by abulte — COMMENTED** [review]
What about using the `action` of the form?

**Comment by maudetes** [comment]
> What about using the `action` of the form?

I first thought about using `action`, but we don't particularly want this behavior on `/fr/datasets` page and I couldn't find a clean way to set these attributes conditionally, I can give it another go !

**Comment by abulte** [comment]
> > What about using the `action` of the form?
> 
> I first thought about using `action`, but we don't particularly want this behavior on `/fr/datasets` page and I couldn't find a clean way to set these attributes conditionally, I can give it another go !

It's ok, we're in a JS component anyway, so it doesn't really matter, except maybe for a11n, we'll see ;-)

File: theme/js/components/search/search-input.vue [hunk 2]
```diff
@@ -77,9 +78,9 @@ export default {
         preventScroll: true,
       });
     },
-    onSubmit() {
-      if (this.redirectUrl)
-        window.location.href = this.redirectUrl;
+    onSubmit(event) {
+      if (!this.submitUrl)
+        event.preventDefault();
```
Comment (abulte) [inline theme/js/components/search/search-input.vue:83]: 🦊

**Review by abulte — APPROVED** [review]

**Pull request merged**